portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +26,24 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> comments=new ArrayList<String>();
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment=request.getParameter("comment");
+    comments.add(comment);
+    response.sendRedirect("/index.html");
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    
+    Gson gson = new Gson();
+    
+    String json=gson.toJson(comments);
+    
+    response.setContentType("application/json;");
+
+    response.getWriter().println(json);
```

Comment:
response.getWriter().println(new Gson().toJson(comments));

If not necessary (e.g. for reuse), we usually not to declare so that readers don't need to go and find how the var is defined.